cli/storage/storage.go

```diff
@@ -52,23 +51,13 @@ func CacheDir() (string, error) {
 	return cacheDir, nil
 }
 
-func repoRootPath() (string, error) {
-	ws, err := workspace.Path()
+var repoRootPath = sync.OnceValues(func() (string, error) {
+	dir, err := exec.Command("git", "rev-parse", "--show-toplevel").CombinedOutput()
 	if err != nil {
-		return "", err
+		return "", fmt.Errorf("failed to run git rev-parse: %s", err)
 	}
-	dotgit, err := os.Stat(filepath.Join(ws, ".git"))
-	if err != nil {
-		if os.IsNotExist(err) {
-			return "", fmt.Errorf("not inside a git repository")
-		}
-		return "", fmt.Errorf("failed to determine git repo root: %s", err)
-	}
-	if dotgit.Mode().IsDir() {
```

Comment:
Will `.git` be a symlink in this case? wondering if we can just check if it's a dir or a symlink.

Where possible I would like to avoid invoking `git` because all of these subprocess invocations add up and lead to more latency on every command invocation compared to bazel.

Reply:
```
> git worktree add ../bb-wt    Preparing worktree (new branch 'bb-wt')                                                   HEAD is now at 70811560f fix store case                                                   > cd ../bb-wt

> cat .git                                                   gitdir: /home/nb/work/buildbuddy/buildbuddy/.git/worktrees/bb-wt                          
                                                                              > ls -al .git
.rw-rw-r-- 65 nb  2 Apr 15:25 .git                                                        

> readlink .git
exit 1
```

It's a file with content pointing to the worktree dir inside   the actual .git dir. https://git-scm.com/docs/git-worktree#_details

Interestingly, it's possible for each worktree to have a separate git-config value 🤔

Reply:
hm ok, I guess we could parse that file but maybe not worth it for now.